completion/all.go

```diff
@@ -13,11 +16,16 @@ import (
 	"github.com/coder/serpent"
 )
 
+const (
+	completionStartTemplate = `# ============ BEGIN {{.Name}} COMPLETION ============`
+	completionEndTemplate   = `# ============ END {{.Name}} COMPLETION ==============`
+)
+
 type Shell interface {
 	Name() string
 	InstallPath() (string, error)
-	UsesOwnFile() bool
 	WriteCompletion(io.Writer) error
+	ProgramName() string
```

Comment:
Kind of annoying, but we need the program name during `InstallCompletion` to figure out the exact header we're looking for - otherwise you obviously couldn't have multiple `serpent` completion scripts installed.